<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>720</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>1280</width>
    <height>720</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1280</width>
    <height>720</height>
   </size>
  </property>
  <property name="baseSize">
   <size>
    <width>1280</width>
    <height>720</height>
   </size>
  </property>
  <property name="contextMenuPolicy">
   <enum>Qt::ContextMenuPolicy::NoContextMenu</enum>
  </property>
  <property name="acceptDrops">
   <bool>false</bool>
  </property>
  <property name="windowTitle">
   <string>Sorting Algorithms Visualizer</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="startButton">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Start</string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>60</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Bubble Sort</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Selection Sort</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Insertion Sort</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Merge Sort</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Heap Sort</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Quick Sort</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Counting Sort</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Radix Sort</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Cocktail Sort</string>
     </property>
    </item>
   </widget>
   <widget class="QRadioButton" name="ascendingRadioButton">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>90</y>
      <width>91</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Ascending</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">orderButtonGroup</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="descendingRadioButton">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>110</y>
      <width>91</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Descending</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">orderButtonGroup</string>
    </attribute>
   </widget>
   <widget class="QLabel" name="textLabel">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>10</y>
      <width>1121</width>
      <height>671</height>
     </rect>
    </property>
    <property name="text">
     <string>TextLabel</string>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignTop</set>
    </property>
   </widget>
   <widget class="QSpinBox" name="elementsCount">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>160</y>
      <width>91</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>2</number>
    </property>
    <property name="maximum">
     <number>10000</number>
    </property>
    <property name="value">
     <number>50</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="delay">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>210</y>
      <width>91</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>10000</number>
    </property>
    <property name="value">
     <number>1</number>
    </property>
   </widget>
   <widget class="QLabel" name="elementsCountLabel">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>140</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Elements Count:</string>
    </property>
   </widget>
   <widget class="QLabel" name="delayLabel">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>190</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Delay (ms):</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="stepByStepRadioButton">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>260</y>
      <width>91</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Step-by-step</string>
    </property>
    <property name="autoExclusive">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">modeButtonGroup</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="continuousRadioButton">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>240</y>
      <width>91</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Continuous</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <property name="autoExclusive">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">modeButtonGroup</string>
    </attribute>
   </widget>
   <widget class="QPushButton" name="nextStepButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>290</y>
      <width>91</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Next Step -&gt;</string>
    </property>
   </widget>
   <widget class="QLabel" name="themeLabel">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>320</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Theme:</string>
    </property>
   </widget>
   <widget class="QComboBox" name="themeComboBox">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>340</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Default</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Cyber</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Autumn</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Night Glow</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Retro Pop</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Solar Burst</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Zen Garden</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Tropical Drift</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Monotone</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Bold Sunset</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Frosty Chill</string>
     </property>
    </item>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="orderButtonGroup"/>
  <buttongroup name="modeButtonGroup"/>
 </buttongroups>
</ui>
